### PR TITLE
Add create mapping endpoint

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

This PR adds the create mapping endpoint

## Related issues

related to #22392

### CODE REVIEW

**Review by christinaausley — APPROVED** [review]
No grammatical adjustments on my end 👍

File: service/src/main/java/io/camunda/service/MappingServices.java [hunk 1]
```diff
@@ -75,5 +75,5 @@ public Optional<MappingEntity> findMapping(final Long mappingKey) {
         .findFirst();
   }
 
-  public record MappingDTO(Long mappingKey, String claimName, String claimValue) {}
+  public record MappingDTO(String claimName, String claimValue) {}
```
Comment (emulbreh) [inline service/src/main/java/io/camunda/service/MappingServices.java:78]: We keep the key in the DTO for users, primarily for updates as far as I can tell. I think it's actually nicer to keep the key outside so it doesn't have to be nullable, but let's make it consistent across all entities.

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -2201,6 +2200,50 @@ paths:
             application/problem+json:
               schema:
                 $ref: "#/components/schemas/ProblemDetail"
+
+  /mapping-rules:
+    post:
+      tags:
+        - Mapping
```
Comment (emulbreh) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2207]: Let's make this "Mapping Rule" for clarity.

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java [hunk 3]
```diff
@@ -325,6 +328,13 @@ public static Either<ProblemDetail, UserDTO> toUserDTO(
                 passwordEncoder.encode(request.getPassword())));
   }
 
+  public static Either<ProblemDetail, MappingDTO> toMappingRequest(
```
Comment (emulbreh) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java:331]:
```suggestion
  public static Either<ProblemDetail, MappingDTO> toMappingDTO(
```
for consistency with toUserDTO().

File: zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/MappingControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+  @MockBean private MappingServices mappingServices;
+
+  @BeforeEach
+  void setup() {
+    when(mappingServices.withAuthentication(any(Authentication.class))).thenReturn(mappingServices);
+  }
+
+  @Test
+  void createMappingShouldReturnAccepted() {
```
Comment (emulbreh) [inline zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/MappingControllerTest.java:42]:
```suggestion
  void createMappingShouldReturnCreated() {
```

File: service/src/main/java/io/camunda/service/MappingServices.java (hunk 1 above)
Comment (emulbreh) [reply service/src/main/java/io/camunda/service/MappingServices.java:78]: This isn't from this PR, but I think it'd be good to standardize the name of these objects to either "mapping rule" or just "mapping". We don't have to fix it in this PR though.

**Review by emulbreh — APPROVED** [review]

File: service/src/main/java/io/camunda/service/MappingServices.java (hunk 1 above)
Comment (nicpuppa) [reply service/src/main/java/io/camunda/service/MappingServices.java:78]: > We keep the key in the DTO for users, primarily for updates as far as I can tell. I think it's actually nicer to keep the key outside so it doesn't have to be nullable, but let's make it consistent across all entities.

It's not in the scope of this PR, so I'll create a follow-up for this. Same for the second comment

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java (hunk 3 above)
Comment (nicpuppa) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java:331]: We should create a follow-up also for this case, we have a mix of `DTO` and `request` object. Can you do this ?

**Pull request merged**